components/vfs/manager.cpp

```diff
@@ -57,6 +57,14 @@ namespace VFS
             (*it)->listResources(mIndex, mStrict ? &strict_normalize_char : &nonstrict_normalize_char);
     }
 
+    void Manager::rebuildIndex()
+    {
+        for (std::vector<Archive*>::const_iterator it = mArchives.begin(); it != mArchives.end(); ++it)
+            (*it)->reset();
+
+        buildIndex();
+    }
```

Comment:
I feel like the additions to the VFS component are kinda redundant, since they do pretty much the same thing as deleting the VFS and creating a new one (which we can already do). How much work would it be to do that and pass a new VFS pointer to all users? If that's too much work, a middle-ground would be to clear the contained archives and add new ones.